### PR TITLE
Updates ready for MQ V9.0.4

### PR DESCRIPTION
**Pull request by arthurbarr**
* Updated to MQ version 9.0.4.0
* Updated to Go version 9
* Removed packages `curl`, `ca-certificates`, and their dependencies, which were only used at build time
* Improved logging
* Helm charts now work on Kubernetes V1.6
* Production Helm chart now includes a default image repository and tag
* Updated to use multi-stage Docker build, so that Go code is built inside a container

### CODE REVIEW

File: charts/ibm-mqadvanced-server-dev/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ This chart deploys a single IBM MQ Advanced for Developers server (queue manager
 
 ## Prerequisites
 
-- Kubernetes 1.7 or greater, with beta APIs enabled
+- Kubernetes 1.6 or greater, with beta APIs enabled
```
Comment (parrobe) [inline charts/ibm-mqadvanced-server-dev/README.md]: Lowered the Kubernetes requirement?

File: charts/ibm-mqadvanced-server-dev/templates/stateful-set.yaml [hunk 2]
```diff
@@ -24,8 +24,10 @@ metadata:
 spec:
   serviceName: {{ .Values.service.name }}
   replicas: 1
+  {{- if and (ge (.Capabilities.KubeVersion.Major | int) 1) (ge (.Capabilities.KubeVersion.Minor | int) 7) }}
```
Comment (parrobe) [inline charts/ibm-mqadvanced-server-dev/templates/stateful-set.yaml]: This looks like a check to make sure the Kube version has a minor of 1.7 but the above line in the readme says 1.6 now.

File: charts/ibm-mqadvanced-server-prod/README.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ This chart deploys a single IBM MQ Advanced server (queue manager) into an IBM C
 
 ## Prerequisites
 
-- Kubernetes 1.7 or greater, with beta APIs enabled
+- Kubernetes 1.6 or greater, with beta APIs enabled
```
Comment (parrobe) [inline charts/ibm-mqadvanced-server-prod/README.md]: Same as above

File: charts/ibm-mqadvanced-server-prod/templates/stateful-set.yaml [hunk 4]
```diff
@@ -24,8 +24,10 @@ metadata:
 spec:
   serviceName: {{ .Values.service.name }}
   replicas: 1
+  {{- if and (ge (.Capabilities.KubeVersion.Major | int) 1) (ge (.Capabilities.KubeVersion.Minor | int) 7) }}
```
Comment (parrobe) [inline charts/ibm-mqadvanced-server-prod/templates/stateful-set.yaml]: Same as above

File: test/kubernetes/helm_test.go [hunk 5]
```diff
@@ -40,6 +40,7 @@ func TestHelmPredefinedVolume(t *testing.T) {
 func TestHelmStorageClass(t *testing.T) {
 	cs := kubeLogin(t)
 	release := strings.ToLower(t.Name())
+	assertKubeVersion(t, cs, 1, 6)
```
Comment (parrobe) [inline test/kubernetes/helm_test.go]: Looks for 1.6 here but a different check looks for 1.7?

**Comment by arthurbarr** [comment]
That's the point.  It only does the rolling update if the version is 1.7 or greater.  Thus allowing the chart to work on Kubernetes 1.6.

**Pull request merged**